packages/govuk-frontend/src/govuk/errors/index.mjs

```diff
@@ -22,6 +22,13 @@ export class GOVUKFrontendError extends Error {
   name = 'GOVUKFrontendError'
 }
 
+/**
+ * Indicates that a Component's required HTML element is missing
+ */
+export class MissingElementError extends GOVUKFrontendError {
+  name = 'MissingElementError'
+}
```

Comment:
Not a blocker, but it's triggered a discussion

To keep file size down, could we use Error `{ cause }` to keep our custom errors to a minimum? Otherwise our published package will grow as custom errors are added

E.g. Using [**TypeError** "…not of the expected type"](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError) for `null` or missing things

### Component errors
Here's an example with nested `{ cause }` to make the error more informative:

```mjs
throw new ComponentError('I has died', {
  cause: new TypeError("Argh, 'ere be no link target #pirates")
}
```

But using the nested `{ cause }` approach to surface all that useful information. It's handy for monitoring tools like [Sentry](https://sentry.io/welcome/) which links chained errors + stack traces

For extra user friendliness a custom `toString()` (see below) could log the component name:

>ComponentError: "Skip link: I has died"
> → Cause: TypeError: "Argh, 'ere be no link target #pirates"

```suggestion
export class ComponentError extends GOVUKFrontendError {
  name = 'ComponentError'
  component = 'Component name'

  toString() {
    return `${this.component}: ${this.message}`;
  }
}
```

Reply:
There's also a gotcha

We'd need to confirm Babel transforms or a polyfill for [JavaScript built-in: Error: cause](https://caniuse.com/mdn-javascript_builtins_error_cause)